pkg/apis/sources/v1alpha1/types_test.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	"testing"
+)
+
+func TestOutputFormatTypeValid(t *testing.T) {
+	tests := []struct {
+		o    OutputFormatType
+		want bool
+	}{
+		{"structured", true},
+		{"binary", true},
+		{"trinary", false},
+		{"quantum", false},
+		{"http", false}, // all events are over HTTP
+		{"messenger_pigeon", false},
```

Comment:
this would be a super fast encoding.

Reply:
All sources starting 0.9 have to support messenger pigeon